### ISSUE TITLE
Z Axis Steps Per MM Limit

### ISSUE THREAD

**Issue by specky532** [markwal/GpxUi#3]
Had great success with this software but need to increase the steps/mm in the z axis past 1000.

Why is there a limit on this in the machineeditor.ui? I don't see that there is any reason for this? About to clone and change this variable to rebuild for myself but I figure this is an issue that other people may also have!

Cheers for making such a great app!

**Comment by markwal** [comment]
Hmmm...  I'm not sure I remember.   I'll see if I can jog my memory by looking at some sources.

**Comment by markwal** [comment]
I think it may have been arbitrary.  However, since GPX currently computes steps by doing floating point math and multiplying the gcode mm by the steps per mm, you probably aren't gonna get the precision you might expect.

You could test this out by editing the .ini file directly.

**Comment by specky532** [comment]
Heya,

I actually did end up just editing the .ini file directly as I didn't have time to set up my environment to recompile. Doing this worked perfectly, at least with the calibration cube that I printed.

**Comment by markwal** [comment]
OK. Cool.

**Comment by markwal** [comment]
Further investigation:  The default maximum for a QDoubleSpinBox in Qt is 100.  As far as I can tell, the control doesn't allow for a "no maximum" case, so I must've picked 1000 as being an order of magnitude away from the typical.  Perhaps the easy fix is to just adjust it to 10000?

**Comment by specky532** [comment]
I would agree, in the case of my printer it only needed to be adjusted up to 1600 (and only in the z) so 10,000 should/would be more than enough.
